rbd/admin/task.go

```diff
@@ -0,0 +1,141 @@
+// +build !nautilus
+
+package admin
+
+import (
+	ccom "github.com/ceph/go-ceph/common/commands"
+	"github.com/ceph/go-ceph/internal/commands"
+)
+
+// TaskAdmin encapsulates management functions for
+// ceph rbd task operations.
+type TaskAdmin struct {
+	conn ccom.MgrCommander
+}
+
+// Task returns a TaskAdmin type for
+// managing ceph rbd task operations.
+func (ra *RBDAdmin) Task() *TaskAdmin {
+	return &TaskAdmin{conn: ra.conn}
+}
+
+// TaskRefs contains the action name and information about the image.
+type TaskRefs struct {
+	Action        string `json:"action"`
+	PoolName      string `json:"pool_name"`
+	PoolNamespace string `json:"pool_namespace"`
+	ImageName     string `json:"image_name"`
+	ImageID       string `json:"image_id"`
+}
+
+// TaskResponse contains the information about the task added on an image.
+type TaskResponse struct {
+	Sequence      int      `json:"sequence"`
+	ID            string   `json:"id"`
+	Message       string   `json:"message"`
+	Refs          TaskRefs `json:"refs"`
+	InProgress    bool     `json:"in_progress"`
+	Progress      float64  `json:"progress"`
+	RetryAttempts int      `json:"retry_attempts"`
+	RetryTime     string   `json:"retry_time"`
+	RetryMessage  string   `json:"retry_message"`
+}
+
+func parseTaskResponse(res commands.Response) (TaskResponse, error) {
+	var taskResponse TaskResponse
+	err := res.NoStatus().Unmarshal(&taskResponse).End()
+
+	return taskResponse, err
+}
+
+func parseTaskResponseList(res commands.Response) ([]TaskResponse, error) {
+	var taskResponseList []TaskResponse
+	err := res.NoStatus().Unmarshal(&taskResponseList).End()
+
+	return taskResponseList, err
+}
+
+// AddFlatten adds a task to flatten a cloned image asynchronously in the background
```

Comment:
I'm probably being overly nitipicky about this but "asynchronously in the background" feels a bit like a statement from the Department of Redundancy Department :-)
Perhaps saying "adds a background task to flatten..." or "asychronous server-side job" or something along those lines?